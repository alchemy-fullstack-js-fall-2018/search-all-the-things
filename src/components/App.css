#landingPage {
  display: flex;
  flex-direction: column;
  align-items: center;

  #searchBar {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    border: 3px solid black;
    background: lightblue;
    padding: 10px;
    width: 75vw;
  }
}

h1 {
  margin-bottom: 0px;
}
