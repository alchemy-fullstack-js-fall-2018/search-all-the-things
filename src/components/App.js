import React, { Component, Fragment } from 'react';
import Paging from './paging/Paging';
import Results from './results/Results';
import Search from './search/Search';
import News from './news/News';
import styles from './App.css';
import { getNews } from './services/newsApi';

export default class App extends Component {

  state = {
    currentPage: 1,
    totalResults: 1,
    searchResults: ''
  };

  updatePage = page => {
    this.setState({ currentPage: page });
  };

  updateTotalResults = totalResults => {
    this.setState({ totalResults });
  };


  updateSearchResults = ({ target }) => {
    this.setState({ [target.name]: target.value, currentPage: 1 });
  };

  render() {
    const { currentPage, searchResults, totalResults } = this.state;
    return (
      <Fragment>
        <div className={styles.header}>
          <h1>Search The News</h1>
          <Search updateSearchResults={this.updateSearchResults} searchResults={searchResults}/><br/>
          <Paging currentPage={currentPage} totalResults={totalResults} updatePage={this.updatePage} />
        </div>
        <Results
          currentPage={currentPage}
          getResults={getNews}
          searchResults={searchResults}
          updateTotalResults={this.updateTotalResults}
          ResultComponent={News}
        />
      </Fragment>
    );
  }
}
