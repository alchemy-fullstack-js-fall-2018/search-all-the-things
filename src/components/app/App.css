.main {
  display: flex;
  flex-direction: column;
  height: 100vh;
  justify-content: space-between;
  text-align: center;
  margin: 0;
}
