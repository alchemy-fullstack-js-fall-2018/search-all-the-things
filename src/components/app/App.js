import React from 'react';
import Pokemons from '../pokemon/Pokemon';

export default function app() {
  return (
    <Pokemons />
  );
}
