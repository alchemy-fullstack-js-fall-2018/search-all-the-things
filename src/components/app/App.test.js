import App from './App';

describe('App Component', () => {
  it('should be a app', () => {
    const app = new App();
    expect(app.render()).toMatchSnapshot();
  });
});
