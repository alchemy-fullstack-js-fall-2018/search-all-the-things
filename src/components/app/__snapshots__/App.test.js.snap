// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should be a app 1`] = `
<main
  className="main"
>
  <Header />
  <SearchBox
    onChange={[Function]}
    onPageSize={[Function]}
    onSearch={[Function]}
    pageSize={20}
    query=""
  />
  <Footer />
</main>
`;
